src/HSE/All.hs

```diff
@@ -73,11 +73,11 @@ parseModuleEx flags file str = do
             ParseOk (x, cs) -> return $ Right (applyFixity fixity x, cs)
             ParseFailed sl msg -> do
                 -- figure out the best line number to grab context from, by reparsing
-                flags <- return $ parseFlagsNoLocations flags
-                ppstr2 <- runCpp (cppFlags flags) file str
-                pe <- return $ case parseFileContentsWithMode (mode flags) ppstr2 of
-                    ParseFailed sl2 _ -> context (srcLine sl2) ppstr2
-                    _ -> context (srcLine sl) ppstr
+                let flags' = parseFlagsNoLocations flags
```

Comment:
I think this is a perfectly legitimate use of `return`, so I'll probably modify hlint not to suggest `let` in case of variable shadowing.